src/start/AuthScreen.js

```diff
@@ -19,52 +19,52 @@ import { generateOtp, openBrowser, closeBrowser } from './oauth';
 import { loginSuccess, navigateToDev, navigateToPassword } from '../actions';
 
 type AuthenticationMethodDetails = {|
-  method: string,
   name: string,
+  displayName: string,
   Icon: IconType,
   handler: string,
 |};
```

Comment:
Given the commit message, it sounds like `'id'` or `'methodName'` might be a better key than `'name'`.

(Either way, these two items could use docstrings, or at least docstringishly descriptive comments.)

Reply:
Yeah, I agree `name` is ambiguous on identifier-like name vs. display name. I think it helps that the other name in the pair is unambiguous, though.

To me `id` would suggest a numeric ID -- like our user IDs, realm IDs, stream IDs, etc. Realms and streams have identifier-like names, even, but then also "IDs" which are numbers. I think `methodName` doesn't add any information beyond `name`; to the extent an auth method's "name" could mean its display name, "method name" could just as well. So I don't think either of those would help.

Some jsdoc is a good idea. I'll add some.

Reply:
> To me `id` would suggest a numeric ID -- like our user IDs, realm IDs, stream IDs, etc.

Ah, that's understandable. And noted. (The meaning I was thinking of was from the DOM:  `document.getElementById('blorple')`, etc.)

Reply:
> (The meaning I was thinking of was from the DOM: `document.getElementById('blorple')`, etc.)

Ah, makes sense! Yeah, within Zulip I think we pretty consistently say "ID" only for a numeric ID.

The one exception I can think of is within the server codebase -- we've managed to keep it out of the API -- where `Realm.string_id` is an identifier-style string, better known as the "subdomain".